e2e_tests/tests/experiment/noop.py

```diff
@@ -0,0 +1,173 @@
+"""
+A python API for creating tests that do no training, but only take the actions they are told to.
+"""
+
+import base64
+import json
+import logging
+import pathlib
+from typing import Any, Dict, Iterable, List, Optional, Union
+
+from determined.common import api
+from determined.common.api import bindings
+from determined.experimental import client
+from tests import config as conf
+
+
+class Exit:
+    def __init__(self, code: int = 0) -> None:
+        self.code = code
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"action": "exit", "code": self.code}
+
+
+class Sleep:
+    def __init__(self, time: float) -> None:
+        self.time = time
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"action": "sleep", "time": self.time}
+
+
+class Report:
+    def __init__(self, metrics: Dict[str, Any], group: str = "training") -> None:
+        self.metrics = metrics
+        self.group = group
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"action": "report", "group": self.group, "metrics": self.metrics}
+
+
+class Checkpoint:
+    def to_dict(self) -> Dict[str, Any]:
+        return {"action": "checkpoint"}
+
+
+class Log:
+    def __init__(self, msg: str, level: int = logging.INFO) -> None:
+        # Transfer message as base64 to allow weird characters, and also so that when searching logs
+        # you don't see the msg string anywhere in "metadata", like expconf.
+        self.base64 = base64.b64encode(msg.encode("utf8")).decode("utf8")
+        self.level = level
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"action": "log", "base64": self.base64, "level": self.level}
+
+
+class CompleteSearcherOperation:
+    def __init__(self, metric: float) -> None:
+        self.metric = metric
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"action": "complete_searcher_operation", "metric": self.metric}
+
+
+Action = Union[Exit, Sleep, Report, Checkpoint, Log, CompleteSearcherOperation]
+
+
+def merge_config(old: Any, new: Any) -> Any:
+    if isinstance(old, dict):
+        out = dict(old)
+        for key, new_value in new.items():
+            if key in out:
+                out[key] = merge_config(out[key], new_value)
+            else:
+                out[key] = new_value
+        return out
+    return new
+
+
+def generate_config(
+    *actions: Action,
```

Comment:
why can't this just be a list? unpacking introduces unnecessary complexity in the function signature IMO, i have to think about whether the args are named, positional, etc

looking at the tests, seems like most of them either pass in a single action, or call `trainlike_actions` to them anyway.

Reply:
meant to comment this on create_experiment actually but same idea

Reply:
yeah it prolly should be a list.  I always like the idea of `*args` but I can't say I always like how it turns out.

I'll make that change.

Reply:
fixed.